addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -774,13 +774,14 @@ var SnippetEditor = Widget.extend({
      * @param {Event} ev
      */
     _onUserValueWidgetRequest: function (ev) {
-        for (const key of Object.keys(this.styles)) {
-            const widget = this.styles[key].findWidget(ev.data.name);
+        const widgets = [];
+        for (const style of Object.values(this.styles)) {
+            const widget = style.findWidget(ev.data.name);
             if (widget) {
-                ev.data.onSuccess(widget);
-                return;
+                widgets.push(widget);
             }
         }
+        ev.data.onSuccess(widgets);
```

Comment:
@sdegueldre It was a mistake to let that event propagate in fact, I always only want to check widgets in other options of the same snippet editor, not its parent (correct me if I am wrong?). The same way, I only update the UI of widgets in other options of the same snippet editor when one of the option changes.

It would be a mess to target a widget from a parent snippet editor since you cannot be sure if another level with other snippet options with the same name will not appear anywhere in the parent chain.

If ok, let's simply add a `ev.stopPropagation` here, maybe in 13.1/13.2.

Reply:
That works too. Adapted the code here, I'll make a PR for it tomorrow in 13.2 and rebase this one on it once it's merged :+1:

Reply:
odoo/odoo#46707